docs/debugging.md

```diff
@@ -0,0 +1,223 @@
+3. Once the Application stops, those variable(s) are now accessible as normal.
+
+The details of this pattern are further explained below.
+
+### Application.run() stop conditions
+
+In a production setting, `Application.run()` should be called only once with no
+arguments, which means it will run indefinitely (until it encounters an error or 
+is manually stopped by the user).
+
+However, for debugging, the following kwargs can be passed to stop the `Application` 
+when the applicable condition is met:
+
+- `timeout`: the given amount of time in seconds to run after initial rebalance or recovery (default: `0.0` == infinite)
+- `count`: number of messages to process from main SDF input topics (default `0` == infinite)
+
+> **NOTE:** If both are passed, the first condition met will trigger the stop.
+
+#### Multiple Application.run() calls
+
+It is safe to do subsequent `Application.run()` calls (even with
+new arguments!); it will simply pick up where it left off. 
+
+There is no need to do any manual cleanup when finished; each run cleans up after itself.
+
+> **NOTE:** You do not need to re-execute the entire Application setup, just `.run()`.
+
+If the Application seems to take longer than desired with getting initial messages or 
+you know there will be a delay, you can also specify it to wait to start counting the 
+timeout until after a first message is received by specifying 
+`timeout_starts_on_first_message=True`.
+
+### ListSink
+
+`ListSink` primarily exists to use alongside `Application.run()` stop conditions.
+
+It collects data where it's used, which can then be interacted with like
+a list once the `Application` stops.
+
+#### Using ListSink
+
+To use a `ListSink`, simply store one in a separate variable and pass it
+like any other `Sink`:
+
+```
+from quixstreams import Application
+from quixstreams.sinks.core.list import ListSink
+
+app = Application(broker_address="localhost:9092")
+topic = app.topic("some-topic")
+list_sink = ListSink()  # sink will be a list-like object
+sdf = app.dataframe(topic=topic).sink(list_sink)
+app.run(timeout=10)  # run for 10 seconds
+```
+
+You can then interact with it once the `Application` stops:
+
+```shell
+> print(list_sink) 
+[{"thing": "x"}, {"thing": "y"}, {"thing": "z"}]
+
+> list_sink[0]
+{"thing": "x"}
+```
+
+> **NOTE:** Though the result is actually a `ListSink`, it behaves like a list...you can even pass it 
+to `Pandas.DataFrame()` and it will work as expected!
+
+
+#### ListSink Limitations
+
+- You can use any number of `ListSink` in an `Application`
+  - each one must have its own variable.
+
+- `ListLink` does not limit it's own size 
+  - Be sure to use it with `Application.run()` stopping conditions.
+
+- `ListSink` does not "refresh" itself per `.run()`; it collects data indefinitely. 
+  - You can remove the current data stored by doing either:
+
+      - `ListSink.clear()` 
+    
+        or...
+    
+      - `ListSink[:] = []`
```

Comment:
Let's simplify this bit, and I also propose to reuse the variable name from previous examples rather than the class name.
```suggestion
  - You can remove the current data stored by doing `list_sink.clear()`.
```